flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/Projection.java

```diff
@@ -39,9 +49,24 @@
     // sealed class
     private Projection() {}
 
-    /** Project the provided {@link DataType} using this {@link Projection}. */
+    /**
+     * Projects a (possibly nested) row data type by returning a new data type that only includes
+     * fields of the given index paths.
+     *
+     * <p>Note on naming: Sometimes, it may get name conflicts when extract fields from the row
```

Comment:
Could we fit the grammar in this doc for a `@PublicEvolving` interface?

Reply:
I reworded this comment to explain it better, wdyt?

Reply:
perfect, thanks :)